datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -3015,6 +3085,18 @@ mod tests {
         make_array(&args).expect("failed to initialize function array")
     }
 
+    fn return_large_array() -> ArrayRef {
+        // Returns: [1, 2, 3, 4]
+        let args = [
+            Arc::new(Int64Array::from(vec![Some(1)])) as ArrayRef,
+            Arc::new(Int64Array::from(vec![Some(2)])) as ArrayRef,
+            Arc::new(Int64Array::from(vec![Some(3)])) as ArrayRef,
+            Arc::new(Int64Array::from(vec![Some(4)])) as ArrayRef,
+        ];
+        let data_type = DataType::Int64;
+        array_array::<i64>(&args, data_type).expect("failed to initialize function array")
+    }
+
```

Comment:
Hack here to create a `LargeList` to avoid spending too much memory for testing.

Reply:
What is the hack?

Reply:
maybe you should use below constructor to construct LargeList
```rust
let list_array = Arc::new(LargeListArray::from_iter_primitive::<Int32Type, _, _>(
        vec![
            Some(vec![Some(0), Some(1), Some(2)]),
            None,
            Some(vec![Some(3), None, Some(5)]),
        ],
    )) as ArrayRef

```